envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/resource/clusters/EnvoyClustersFactory.kt

```diff
@@ -64,9 +70,19 @@ class EnvoyClustersFactory(
         return insecureClusters.map { cluster ->
             val upstreamTlsContext = createTlsContextWithSdsSecretConfig(cluster.name)
             val secureCluster = Cluster.newBuilder(cluster)
-            secureCluster.setTransportSocket(TransportSocket.newBuilder()
-                    .setName("envoy.transport_sockets.tls")
-                    .setTypedConfig(Any.pack(upstreamTlsContext)))
+
+            val matchTlsContext = Cluster.TransportSocketMatch.newBuilder()
```

Comment:
We need to ensure that transport socket match works with our minimal supported Envoy version. 
It's relatively new feature.

Reply:
It works from: 1.14.0-dev-6c2137468c25d167dbbe4719b0ecaf343bfb4233

Does not work for:  1.13.0-dev-b7bef67c256090919a4585a1a06c42f15d640a09.

I'll add CI links once they are finished.

Reply:
https://app.circleci.com/pipelines/github/allegro/envoy-control/836/workflows/fcf5adae-8938-4a48-86fa-ff409694b13c/jobs/2441/tests SAN fails because of a bug.

https://app.circleci.com/pipelines/github/allegro/envoy-control/837/workflows/95049cb9-6dd9-43f7-8e32-2a712184c16f/jobs/2443/tests - container does not start